src/Perl6/Compiler.nqp

```diff
@@ -7,6 +7,31 @@ class Perl6::Compiler is HLL::Compiler {
     has $!language_modifier; # Active language modifier; PREVIEW mostly.
     has $!language_revisions; # Hash of language revision letters. See gen/<vm>/main-version.nqp
     has $!can_language_versions; # List of valid language version
+    has $!rakudo-home;
+
+    method BUILD() {
+        # Determine Perl6 and NQP dirs.
+#?if jvm
+        my $sep := nqp::atkey(nqp::jvmgetproperties,'os.name') eq 'MSWin32' ?? '\\' !! '/';
+        my $execname := nqp::atkey(nqp::jvmgetproperties,'perl6.execname');
+#?endif
+#?if !jvm
+        my $config := nqp::backendconfig();
+        my $sep := $config<osname> eq 'MSWin32' ?? '\\' !! '/';
```

Comment:
If we gonna have pieces like this scattered all around it's no good. Feels to me like we should expose some of the most required platform-dependant variable from nqp-configure via hll-config to reduce the amounts of platform-detecting code.

This is not a request for change, just something to consider.